docs/design/blocks.md

```diff
@@ -0,0 +1,645 @@
+# Schema Blocks
+
+The framework currently implements nested attribute support in schemas that is only available in the recent Terraform Plugin Protocol version 6. Prior versions of the protocol and SDK implemented support for blocks. Provider developers may have previously existing resources from the older Terraform Plugin SDK that they wish to mux with or bring into the new framework without introducing breaking changes for practitioners.
+
+This design documentation will walk through and recommend options for schema block handling in the framework.
+
+## Background
+
+### Terraform Configuration Language
+
+Terraform's configuration language is built on top of [HashiCorp Configuration Language (HCL)](https://github.com/hashicorp/hcl) and conceptually supports two syntaxes for declaring configuration as described in the [Configuration Syntax documentation](https://www.terraform.io/docs/language/syntax/configuration.html):
+
+- **Attributes** (**Arguments**): Assign a value to a particular name.
+- **Blocks**: Container for other content.
+
+With the attributes using an equals sign (`=`), such as:
+
+```terraform
+resource "example_thing" "example" {
+  some_attr = "abc123" # Example attribute/argument
+}
+```
+
+While blocks are structural and defined without an equals sign (`=`), such as:
+
+```terraform
+resource "example_thing" "example" {
+  some_block {
+    # ... attributes/arguments ...
+  }
+}
+```
+
+Blocks can be repeated multiple times if the underlying schema type allows multiple declarations, such as:
+
+```terraform
+resource "example_thing" "example" {
+  some_block {
+    # ... attributes/arguments ...
+  }
+
+  some_block {
+    # ... attributes/arguments ...
+  }
+}
+```
+
+To handle situations where the block configuration is not static, the configuration language added support for [`dynamic` block expressions](https://www.terraform.io/docs/language/expressions/dynamic-blocks.html). In contrast, working with attributes is relatively simpler as there are many expressions that can operate directly on any collection value, such as [`for` expressions](https://www.terraform.io/docs/language/expressions/for.html).
+
+Blocks also have other limitations in Terraform, such as not supporting sensitivity. Instead, each nested attribute must implement sensitivity as appropriate.
+
+### Terraform Plugin Protocol
+
+In terms of the protocol, all configuration information from the plugin is referred to as the schema. Prior to the recent version 6 of the protocol, only blocks were supported implementing complex schema types.
```

Comment:
```suggestion
In terms of the protocol, all configuration information from the plugin is referred to as the schema. Prior to the recent version 6 of the protocol, implementing complex schema types could only be done using blocks.
```

Perhaps also add a brief sentence pointing out the addition of `nested_type` to `Attribute` and what exactly this enables.

Reply:
Thanks! Pulled in that change and added a quick sentence there about `nested_type`.